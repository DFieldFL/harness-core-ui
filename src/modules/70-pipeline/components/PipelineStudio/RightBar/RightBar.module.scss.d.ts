/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly rightBar: string
  readonly iconButton: string
  readonly selected: string
  readonly enableGitExpIcon: string
  readonly codebaseIcon: string
  readonly notificationsIcon: string
  readonly flowControlIcon: string
  readonly policySetsIcon: string
  readonly variablesIcon: string
  readonly predefinedValue: string
  readonly gitSyncPopover: string
  readonly labelWrapper: string
}
export default styles
