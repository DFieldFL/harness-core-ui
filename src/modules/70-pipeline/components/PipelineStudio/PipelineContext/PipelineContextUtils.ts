/**
 * TODO: Temporary code for CRUD CI API operations for pipelines - only for stages that contains ci (build) stage
 */

import {
  postPipelinePromise,
  ResponseString,
  Failure,
  PostPipelineQueryParams,
  NgPipelineRequestBody,
  putPipelinePromise,
  PutPipelineQueryParams,
  PutPipelinePathParams,
  ResponseNGPipelineResponse,
  GetPipelineQueryParams,
  GetPipelinePathParams,
  getPipelinePromise
} from 'services/cd-ng'
import { getConfig, getUsingFetch, mutateUsingFetch, GetUsingFetchProps, MutateUsingFetchProps } from 'services/config'

export const postPipelinePromiseCI = (
  props: MutateUsingFetchProps<ResponseString, Failure | Error, PostPipelineQueryParams, NgPipelineRequestBody, void>,
  signal?: RequestInit['signal']
) =>
  mutateUsingFetch<ResponseString, Failure | Error, PostPipelineQueryParams, NgPipelineRequestBody, void>(
    'POST',
    getConfig('ci'),
    `/pipelines`,
    props,
    signal
  )

export const putPipelinePromiseCI = (
  {
    pipelineIdentifier,
    ...props
  }: MutateUsingFetchProps<
    ResponseString,
    Failure | Error,
    PutPipelineQueryParams,
    NgPipelineRequestBody,
    PutPipelinePathParams
  > & { pipelineIdentifier: string },
  signal?: RequestInit['signal']
) =>
  mutateUsingFetch<
    ResponseString,
    Failure | Error,
    PutPipelineQueryParams,
    NgPipelineRequestBody,
    PutPipelinePathParams
  >('PUT', getConfig('ng/api'), `/pipelines/${pipelineIdentifier}`, props, signal)

export const getPipelinePromiseCI = (
  {
    pipelineIdentifier,
    ...props
  }: GetUsingFetchProps<ResponseNGPipelineResponse, Failure | Error, GetPipelineQueryParams, GetPipelinePathParams> & {
    pipelineIdentifier: string
  },
  signal?: RequestInit['signal']
) =>
  getUsingFetch<ResponseNGPipelineResponse, Failure | Error, GetPipelineQueryParams, GetPipelinePathParams>(
    getConfig('ci'),
    `/pipelines/${pipelineIdentifier}`,
    props,
    signal
  )

export function postPipelinePromiseFactory(module: 'ci' | 'cd') {
  if (module === 'ci') {
    return postPipelinePromiseCI
  }
  return postPipelinePromise
}

export function putPipelinePromiseFactory(module: 'ci' | 'cd') {
  if (module === 'ci') {
    return putPipelinePromiseCI
  }
  return putPipelinePromise
}

export function getPipelinePromiseFactory(module: 'ci' | 'cd') {
  if (module === 'ci') {
    return getPipelinePromiseCI
  }
  return getPipelinePromise
}
