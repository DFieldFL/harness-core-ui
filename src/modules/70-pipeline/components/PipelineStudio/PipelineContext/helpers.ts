import type { PipelineStageWrapper } from '@pipeline/utils/pipelineTypes'
import type { PipelineInfoConfig, StageElementConfig, StageElementWrapperConfigConfig } from 'services/cd-ng'

export function getStageFromPipeline<T extends StageElementConfig = StageElementConfig>(
  stageId: string,
  localPipeline: PipelineInfoConfig
): PipelineStageWrapper<T> {
  let stage: StageElementWrapperConfigConfig | undefined = undefined
  let parent: StageElementWrapperConfigConfig | undefined = undefined
  if (localPipeline?.stages) {
    localPipeline.stages?.some?.(item => {
      if (item?.stage && item.stage.identifier === stageId) {
        stage = item
        return true
      } else if (item?.parallel) {
        stage = getStageFromPipeline(stageId, { stages: item.parallel } as unknown as PipelineInfoConfig).stage
        if (stage) {
          parent = item
          return true
        }
      }
    })
  }
  return { stage, parent }
}

export function getStagePathFromPipeline(stageId: string, prefix: string, pipeline: PipelineInfoConfig): string {
  if (Array.isArray(pipeline.stages)) {
    for (let i = 0; i < pipeline.stages.length; i++) {
      const item = pipeline.stages[i]

      if (item?.stage?.identifier === stageId) {
        return `${prefix}.${i}`
      }

      if (item.parallel) {
        return getStagePathFromPipeline(stageId, `${prefix}.${i}.parallel`, {
          stages: item.parallel
        } as PipelineInfoConfig)
      }
    }
  }

  return prefix
}
