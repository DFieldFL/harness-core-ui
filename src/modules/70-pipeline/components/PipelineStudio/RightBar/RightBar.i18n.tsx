export default {
  templateLibrary: 'Template Library',
  inputVariables: 'Input Variables',
  pipelineSave: 'Pipeline started successfully',
  pipelineUnSave: 'Please save the pipeline first',
  runPipelineFailed: 'Failed to run pipeline',
  runPipeline: 'Run Pipeline'
}
