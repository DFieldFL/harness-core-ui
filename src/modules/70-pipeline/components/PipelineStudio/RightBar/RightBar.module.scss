.right-bar {
  background-color: #fafafa;
  z-index: 21;
  box-shadow: -1px 0 2px 0 rgba(0, 0, 0, 0.25);
  display: grid;
  grid-template-rows: 1fr 60px 60px 60px 1fr;

  .icon-button {
    padding: 0px !important;
    position: relative !important;
    border: 0 !important;
    background-color: transparent !important;

    &.selected {
      color: var(--blue-500) !important;
    }
    &.notificationsIcon {
      :global(.bp3-icon) {
        color: var(--blue-500) !important;
      }
    }

    &.variablesIcon {
      &.selected {
        color: #9d7cff !important;
      }
      &:hover {
        color: #9d7cff !important;
      }
      :global(.bp3-icon) {
        color: #9d7cff !important;
      }
    }

    :global(.bp3-icon) {
      position: absolute;
      top: -15px;
    }
  }

  :global(.bp3-popover-wrapper) {
    text-align: center;
  }

  .codebaseConfiguration {
    &.notConfigured {
      > span svg path:last-child {
        // TODO: Replace color once it is available in UIKit
        fill: #ff8800 !important;
      }
    }

    &.valid {
      > span svg path {
        // TODO: Replace color once it is available in UIKit
        fill: #06c270 !important;
      }
    }

    &.invalid {
      // TODO: Replace color once it is available in UIKit
      color: #9293ab !important;
    }

    &.validating {
      // DO NOT REMOVE
      color: inherit;
    }
  }
}
.predefinedValue {
  font-size: 13px;
  color: var(--grey-500);
  margin-top: calc(var(--spacing-small) / -1);
  margin-bottom: var(--spacing-medium);
}
