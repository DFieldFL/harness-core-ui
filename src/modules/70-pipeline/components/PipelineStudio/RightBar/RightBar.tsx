import React from 'react'
import * as Yup from 'yup'
import cx from 'classnames'
import {
  Button,
  IconName,
  Formik,
  FormikForm,
  RUNTIME_INPUT_VALUE,
  TextInput,
  FormInput,
  Text
} from '@wings-software/uicore'
import { useParams } from 'react-router-dom'
import { isEmpty, set } from 'lodash-es'
import { Classes, Dialog } from '@blueprintjs/core'
import flatten from 'lodash-es/flatten'
import { useStrings } from 'framework/strings'
import { useAppStore } from 'framework/AppStore/AppStoreContext'
import {
  ConnectorInfoDTO,
  getConnectorPromise,
  getTestConnectionResultPromise,
  getTestGitRepoConnectionResultPromise,
  PipelineInfoConfig,
  useGetConnector
} from 'services/cd-ng'
import {
  ConnectorReferenceField,
  ConnectorReferenceFieldProps
} from '@connectors/components/ConnectorReferenceField/ConnectorReferenceField'
import {
  getIdentifierFromValue,
  getScopeFromDTO,
  getScopeFromValue
} from '@common/components/EntityReference/EntityReference'
import { useFeatureFlag } from '@common/hooks/useFeatureFlag'
import type { PipelineType } from '@common/interfaces/RouteInterfaces'
import { Scope } from '@common/interfaces/SecretsInterface'
import { PipelineContext } from '../PipelineContext/PipelineContext'
import { DrawerTypes } from '../PipelineContext/PipelineActions'
import { RightDrawer } from '../RightDrawer/RightDrawer'
import { StageTypes } from '../Stages/StageTypes'
import css from './RightBar.module.scss'

interface CodebaseValues {
  connectorRef?: ConnectorReferenceFieldProps['selected']
  repoName?: string
}

enum CodebaseStatuses {
  ZeroState = 'zeroState',
  NotConfigured = 'notConfigured',
  Valid = 'valid',
  Invalid = 'invalid',
  Validating = 'validating'
}

const codebaseIcons: Record<CodebaseStatuses, IconName> = {
  [CodebaseStatuses.ZeroState]: 'codebase-zero-state',
  [CodebaseStatuses.NotConfigured]: 'codebase-not-configured',
  [CodebaseStatuses.Valid]: 'codebase-valid',
  [CodebaseStatuses.Invalid]: 'codebase-invalid',
  [CodebaseStatuses.Validating]: 'codebase-validating'
}

export const RightBar = (): JSX.Element => {
  const {
    state: {
      pipeline,
      pipelineView,
      pipelineView: {
        drawerData: { type }
      }
    },
    updatePipeline,
    updatePipelineView
  } = React.useContext(PipelineContext)
  const isFlowControlEnabled = useFeatureFlag('NG_BARRIERS')
  const codebase = (pipeline as PipelineInfoConfig)?.properties?.ci?.codebase
  const [codebaseStatus, setCodebaseStatus] = React.useState<CodebaseStatuses>(CodebaseStatuses.ZeroState)

  const { accountId, projectIdentifier, orgIdentifier } = useParams<
    PipelineType<{
      orgIdentifier: string
      projectIdentifier: string
      pipelineIdentifier: string
      accountId: string
    }>
  >()

  const [isCodebaseDialogOpen, setIsCodebaseDialogOpen] = React.useState(false)
  const codebaseInitialValues: CodebaseValues = {
    repoName: codebase?.repoName
  }

  const connectorId = getIdentifierFromValue((codebase?.connectorRef as string) || '')
  const initialScope = getScopeFromValue((codebase?.connectorRef as string) || '')

  const { data: connector, loading, refetch } = useGetConnector({
    identifier: connectorId,
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier: initialScope === Scope.ORG || initialScope === Scope.PROJECT ? orgIdentifier : undefined,
      projectIdentifier: initialScope === Scope.PROJECT ? projectIdentifier : undefined
    },
    lazy: true,
    debounce: 300
  })

  const [connectionType, setConnectionType] = React.useState('')
  const [connectorUrl, setConnectorUrl] = React.useState('')

  if (connector?.data?.connector) {
    const scope = getScopeFromDTO<ConnectorInfoDTO>(connector?.data?.connector)
    codebaseInitialValues.connectorRef = {
      label: connector?.data?.connector.name || '',
      value: `${scope !== Scope.PROJECT ? `${scope}.` : ''}${connector?.data?.connector.identifier}`,
      scope: scope,
      live: connector?.data?.status?.status === 'SUCCESS',
      connector: connector?.data?.connector
    }
  }

  React.useEffect(() => {
    if (!isEmpty(codebase?.connectorRef)) {
      refetch()
    }
  }, [codebase?.connectorRef])

  React.useEffect(() => {
    if (connector?.data?.connector) {
      setConnectionType(connector?.data?.connector.spec.type)
      setConnectorUrl(connector?.data?.connector.spec.url)
    }
  }, [
    connector?.data?.connector,
    connector?.data?.connector?.spec.type,
    connector?.data?.connector?.spec.url,
    setConnectionType,
    setConnectorUrl
  ])

  const { selectedProject } = useAppStore()

  const pipelineStages = flatten(pipeline?.stages?.map(s => s.parallel || s))

  const ciStageExists = pipelineStages?.some?.(stage => {
    if (stage?.stage?.type) {
      return stage?.stage?.type === StageTypes.BUILD
    } else {
      return false
    }
  })

  const isCodebaseEnabled =
    typeof codebaseStatus !== 'undefined' &&
    selectedProject?.modules &&
    selectedProject.modules.indexOf?.('CI') > -1 &&
    ciStageExists

  const atLeastOneCloneCodebaseEnabled = pipelineStages?.some?.(stage => (stage?.stage?.spec as any)?.cloneCodebase)

  React.useEffect(() => {
    if (atLeastOneCloneCodebaseEnabled) {
      if (!codebase?.connectorRef) {
        setCodebaseStatus(CodebaseStatuses.NotConfigured)
      } else {
        const validate = async () => {
          setCodebaseStatus(CodebaseStatuses.Validating)

          const connectorResult = await getConnectorPromise({
            identifier: connectorId,
            queryParams: {
              accountIdentifier: accountId,
              orgIdentifier: initialScope === Scope.ORG || initialScope === Scope.PROJECT ? orgIdentifier : undefined,
              projectIdentifier: initialScope === Scope.PROJECT ? projectIdentifier : undefined
            }
          })

          if (connectorResult?.data?.connector?.spec.type === 'Account') {
            try {
              const response = await getTestGitRepoConnectionResultPromise({
                identifier: connectorId,
                queryParams: {
                  accountIdentifier: accountId,
                  orgIdentifier:
                    initialScope === Scope.ORG || initialScope === Scope.PROJECT ? orgIdentifier : undefined,
                  projectIdentifier: initialScope === Scope.PROJECT ? projectIdentifier : undefined,
                  repoURL:
                    (connectorResult?.data?.connector?.spec.url[
                      connectorResult?.data?.connector?.spec.url.length - 1
                    ] === '/'
                      ? connectorResult?.data?.connector?.spec.url
                      : connectorResult?.data?.connector?.spec.url + '/') + codebase?.repoName
                },
                body: undefined
              })

              if (response?.data?.status === 'SUCCESS') {
                setCodebaseStatus(CodebaseStatuses.Valid)
              } else {
                setCodebaseStatus(CodebaseStatuses.Invalid)
              }
            } catch (error) {
              setCodebaseStatus(CodebaseStatuses.Invalid)
            }
          } else {
            try {
              const response = await getTestConnectionResultPromise({
                identifier: connectorId,
                queryParams: {
                  accountIdentifier: accountId,
                  orgIdentifier:
                    initialScope === Scope.ORG || initialScope === Scope.PROJECT ? orgIdentifier : undefined,
                  projectIdentifier: initialScope === Scope.PROJECT ? projectIdentifier : undefined
                },
                body: undefined
              })

              if (response?.data?.status === 'SUCCESS') {
                setCodebaseStatus(CodebaseStatuses.Valid)
              } else {
                setCodebaseStatus(CodebaseStatuses.Invalid)
              }
            } catch (error) {
              setCodebaseStatus(CodebaseStatuses.Invalid)
            }
          }
        }

        validate()
      }
    } else {
      setCodebaseStatus(CodebaseStatuses.ZeroState)
    }
  }, [codebase?.connectorRef, codebase?.repoName, atLeastOneCloneCodebaseEnabled])

  const openCodebaseDialog = React.useCallback(() => {
    setIsCodebaseDialogOpen(true)
  }, [setIsCodebaseDialogOpen])

  const closeCodebaseDialog = React.useCallback(() => {
    setIsCodebaseDialogOpen(false)

    if (!connector?.data?.connector?.spec.type && !connector?.data?.connector?.spec.url) {
      setConnectionType('')
      setConnectorUrl('')
    }
  }, [
    connector?.data?.connector?.spec.type,
    connector?.data?.connector?.spec.url,
    setIsCodebaseDialogOpen,
    setConnectionType,
    setConnectorUrl
  ])

  const { getString } = useStrings()

  return (
    <div className={css.rightBar}>
      {isCodebaseEnabled && (
        <Button
          className={cx(css.iconButton)}
          text={getString('codebase')}
          font={{ weight: 'semi-bold', size: 'xsmall' }}
          icon={codebaseIcons[codebaseStatus] as IconName}
          iconProps={{ size: 20 }}
          minimal
          withoutCurrentColor
          onClick={() => {
            updatePipelineView({
              ...pipelineView,
              isDrawerOpened: false,
              drawerData: { type: DrawerTypes.AddStep },
              isSplitViewOpen: false,
              splitViewData: {}
            })
            openCodebaseDialog()
          }}
        />
      )}

      <Button
        className={cx(css.iconButton, css.notificationsIcon, {
          [css.selected]: type === DrawerTypes.PipelineNotifications
        })}
        onClick={() => {
          updatePipelineView({
            ...pipelineView,
            isDrawerOpened: true,
            drawerData: { type: DrawerTypes.PipelineNotifications },
            isSplitViewOpen: false,
            splitViewData: {}
          })
        }}
        font={{ weight: 'semi-bold', size: 'xsmall' }}
        icon="right-bar-notification"
        iconProps={{ size: 28 }}
        text={getString('notifications.notificationRules')}
        withoutCurrentColor={true}
      />
      {isFlowControlEnabled && (
        <Button
          className={cx(css.iconButton, css.flowControlIcon, {
            [css.selected]: type === DrawerTypes.FlowControl
          })}
          onClick={() => {
            updatePipelineView({
              ...pipelineView,
              isDrawerOpened: true,
              drawerData: { type: DrawerTypes.FlowControl },
              isSplitViewOpen: false,
              splitViewData: {}
            })
          }}
          font={{ weight: 'semi-bold', size: 'xsmall' }}
          icon="settings"
          iconProps={{ size: 20 }}
          text={getString('pipeline.barriers.flowControl')}
        />
      )}

      <Button
        className={cx(css.iconButton, css.variablesIcon, { [css.selected]: type === DrawerTypes.PipelineVariables })}
        onClick={() =>
          updatePipelineView({
            ...pipelineView,
            isDrawerOpened: true,
            drawerData: { type: DrawerTypes.PipelineVariables },
            isSplitViewOpen: false,
            splitViewData: {}
          })
        }
        font={{ weight: 'semi-bold', size: 'xsmall' }}
        icon="pipeline-variables"
        iconProps={{ size: 20 }}
        text={getString('variablesText')}
        data-testid="input-variable"
      />
      <div />
      {isCodebaseDialogOpen && (
        <Dialog
          isOpen={true}
          title={
            // TODO: Move to strings
            codebaseStatus === CodebaseStatuses.NotConfigured ? 'Configure Codebase' : 'Edit Codebase Configuration'
          }
          onClose={closeCodebaseDialog}
        >
          <Formik
            enableReinitialize
            initialValues={codebaseInitialValues}
            validationSchema={Yup.object().shape({
              connectorRef: Yup.mixed().required(getString('fieldRequired', { field: getString('connector') })),
              ...(connectionType === 'Account' && {
                repoName: Yup.string().required(getString('validation.repositoryName'))
              })
            })}
            onSubmit={(values): void => {
              set(pipeline, 'properties.ci.codebase', {
                connectorRef:
                  typeof values.connectorRef === 'string' ? values.connectorRef : values.connectorRef?.value,
                ...(values.repoName && { repoName: values.repoName }),
                build: RUNTIME_INPUT_VALUE
              })

              // Repo level connectors should not have repoName
              if (connectionType === 'Repo' && (pipeline as PipelineInfoConfig)?.properties?.ci?.codebase?.repoName) {
                delete (pipeline as PipelineInfoConfig)?.properties?.ci?.codebase?.repoName
              }

              updatePipeline(pipeline)

              closeCodebaseDialog()
            }}
          >
            {({ values, setFieldValue, submitForm, errors }) => (
              <>
                <div className={Classes.DIALOG_BODY}>
                  <FormikForm>
                    <ConnectorReferenceField
                      name="connectorRef"
                      type={['Git', 'Github', 'Gitlab', 'Bitbucket', 'Codecommit']}
                      selected={values.connectorRef}
                      width={460}
                      error={errors?.connectorRef}
                      label={getString('connector')}
                      placeholder={loading ? getString('loading') : getString('select')}
                      disabled={loading}
                      accountIdentifier={accountId}
                      projectIdentifier={projectIdentifier}
                      orgIdentifier={orgIdentifier}
                      onChange={(value, scope) => {
                        setConnectionType(value.spec.type)
                        setConnectorUrl(value.spec.url)

                        setFieldValue('connectorRef', {
                          label: value.name || '',
                          value: `${scope !== Scope.PROJECT ? `${scope}.` : ''}${value.identifier}`,
                          scope: scope,
                          live: value?.status?.status === 'SUCCESS',
                          connector: value
                        })
                      }}
                    />
                    {connectionType === 'Repo' ? (
                      <>
                        <Text margin={{ bottom: 'xsmall' }}>
                          {getString('pipelineSteps.build.create.repositoryNameLabel')}
                        </Text>
                        <TextInput name="repoName" value={connectorUrl} style={{ flexGrow: 1 }} disabled />
                      </>
                    ) : (
                      <>
                        <FormInput.Text
                          // TODO: Move to strings, in EditStageView too
                          label={'Repository Name'}
                          name="repoName"
                          style={{ flexGrow: 1 }}
                        />
                        {connectorUrl.length > 0 ? (
                          <div className={css.predefinedValue}>
                            {(connectorUrl[connectorUrl.length - 1] === '/' ? connectorUrl : connectorUrl + '/') +
                              (values.repoName ? values.repoName : '')}
                          </div>
                        ) : null}
                      </>
                    )}
                  </FormikForm>
                </div>
                <div className={Classes.DIALOG_FOOTER}>
                  <Button intent="primary" text={getString('save')} onClick={submitForm} /> &nbsp; &nbsp;
                  <Button text={getString('cancel')} onClick={closeCodebaseDialog} />
                </div>
              </>
            )}
          </Formik>
        </Dialog>
      )}
      <RightDrawer />
    </div>
  )
}
