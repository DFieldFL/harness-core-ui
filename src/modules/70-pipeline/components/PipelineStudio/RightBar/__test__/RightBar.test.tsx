import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { factory } from '@pipeline/components/PipelineSteps/Steps/__tests__/StepTestUtil'
import { RightBar } from '../RightBar'
import { PipelineContext, PipelineContextInterface } from '../../PipelineContext/PipelineContext'

jest.mock('services/cd-ng', () => ({
  getConnectorPromise: () => Promise.resolve(connectorMock),
  useGetConnector: () => ({
    loading: false,
    data: connectorMock,
    refetch: jest.fn()
  })
}))

jest.mock('../../RightDrawer/RightDrawer', () => ({
  RightDrawer: () => <div />
}))

const connectorMock = {
  data: {
    connector: {
      name: 'Git5',
      identifier: 'Git5',
      description: '',
      orgIdentifier: 'CV',
      projectIdentifier: 'Milos2',
      tags: {},
      type: 'Git',
      spec: {
        url: 'https://github.com/wings-software/template-yaml-bugbash.git',
        branchName: null,
        delegateSelectors: [],
        type: 'Http',
        connectionType: 'Repo',
        spec: {
          username: 'AutoUserHarness1',
          usernameRef: null,
          passwordRef: 'GitPass2'
        },
        gitSync: {
          enabled: false,
          customCommitAttributes: null,
          syncEnabled: false
        }
      }
    }
  }
}

const stateMock = {
  pipeline: {
    name: 'P1',
    identifier: 'P1',
    projectIdentifier: 'Milos2',
    orgIdentifier: 'CV',
    tags: {},
    properties: {
      ci: {
        codebase: {
          connectorRef: 'Git5',
          build: '<+input>'
        }
      }
    },
    stages: [
      {
        stage: {
          name: 'S1',
          identifier: 'S1',
          description: '',
          type: 'CI',
          spec: {
            cloneCodebase: true,
            execution: {
              steps: []
            },
            serviceDependencies: []
          }
        }
      }
    ]
  },
  originalPipeline: {
    name: 'P1',
    identifier: 'P1',
    projectIdentifier: 'Milos2',
    orgIdentifier: 'CV',
    tags: {}
  },
  pipelineIdentifier: '-1',
  pipelineView: {
    isSplitViewOpen: true,
    isDrawerOpened: false,
    splitViewData: {
      type: 'StageView'
    },
    drawerData: {
      type: 'AddCommand'
    }
  },
  gitDetail: {},
  isLoading: false,
  isBEPipelineUpdated: false,
  isDBInitialized: true,
  isUpdated: true,
  isInitialized: true,
  selectionState: {
    selectedStageId: 'S1'
  },
  error: ''
}

const pipelineContext: PipelineContextInterface = {
  updatePipeline: jest.fn(),
  state: stateMock as any,
  stepsFactory: factory,
  stagesMap: {},
  isReadonly: false,
  setSchemaErrorView: jest.fn(),
  view: 'ui',
  renderPipelineStage: jest.fn(),
  setView: jest.fn(),
  updateGitDetails: jest.fn(),
  updatePipelineView: jest.fn(),
  fetchPipeline: jest.fn(),
  deletePipelineCache: jest.fn(),
  getStageFromPipeline: jest.fn(),
  setYamlHandler: jest.fn(),
  runPipeline: jest.fn(),
  updateStage: jest.fn(),
  pipelineSaved: jest.fn(),
  setSelectedStageId: jest.fn(),
  setSelectedStepId: jest.fn(),
  setSelectedSectionId: jest.fn(),
  setSelection: jest.fn(),
  getStagePathFromPipeline: jest.fn()
}

describe('RightBar', () => {
  test('renders correctly', () => {
    const { container } = render(
      <TestWrapper>
        <PipelineContext.Provider value={pipelineContext}>
          <RightBar />
        </PipelineContext.Provider>
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
