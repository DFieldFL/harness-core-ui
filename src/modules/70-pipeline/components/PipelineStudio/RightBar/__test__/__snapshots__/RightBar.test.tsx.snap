// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightBar renders correctly 1`] = `
<div>
  <div
    class="rightBar"
  >
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main iconButton variablesIcon StyledProps--font StyledProps--font-weight-semi-bold StyledProps--font-size-xsmall Button--withLeftIcon Button--variation Button--variation-tertiary"
      data-testid="input-variable"
      type="button"
    >
      <span
        class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
        data-icon="pipeline-variables"
      >
        <svg
          height="20"
          viewBox="0 0 15.4 12.2"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <path
              d="M12.5 6.1l-.2-.4c-.1-.3-.2-.4-.2-.5l-.2-.5c-.2-.4-.3-.6-.5-.6s-.4.2-.6.7c-.1 0-.2-.1-.2-.2 0-.2.1-.4.4-.7.2-.3.5-.4.6-.4.4 0 .8.5 1.2 1.6l.2.3.1-.2c.7-1.1 1.3-1.7 1.8-1.7.1 0 .3 0 .4.1l-.6.6h-.2c-.4 0-.8.4-1.3 1.3l-.1.2.1.3c.4 1.2.8 1.8 1.1 1.8.3 0 .5-.2.6-.6.1.1.1.1.1.2 0 .2-.1.4-.4.6-.3.2-.5.4-.7.4-.4 0-.8-.5-1.2-1.6l-.1-.4-.1.3c-.7 1.2-1.3 1.8-2 1.8-.2 0-.4-.1-.6-.2l.6-.6c.1.1.2.2.4.2.4 0 .8-.4 1.3-1.1l.3-.4v-.3zm-6.1-.5L2.6.9h3.8s1.3-.1 1.8.6.6 1.8.6 1.8h.5L9.1 0H0v.5l4.7 5.7L0 11.7v.5h9.3l.6-3.6-.5-.1s-.2.8-.8 1.5c-.6.7-3.1.6-3.1.6H2.1l4.3-5z"
              id="pipeline-variables_svg__a"
            />
          </defs>
          <clippath
            id="pipeline-variables_svg__b"
          >
            <use
              overflow="visible"
              xlink:href="#pipeline-variables_svg__a"
            />
          </clippath>
          <path
            clip-path="url(#pipeline-variables_svg__b)"
            d="M-5-5h25.4v22.2H-5z"
            fill="#9d7cff"
          />
        </svg>
      </span>
      <span
        class="bp3-button-text"
      >
        variablesText
      </span>
    </button>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main iconButton notificationsIcon StyledProps--font StyledProps--font-weight-semi-bold StyledProps--font-size-xsmall Button--withLeftIcon Button--variation Button--variation-tertiary"
      type="button"
    >
      <span
        class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
        data-icon="right-bar-notification"
      >
        <svg
          fill="none"
          height="28"
          viewBox="0 0 20 20"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.167 10.05A5.833 5.833 0 0110 4.219v0-.885.885a5.833 5.833 0 015.833 5.833V15H8.765h2.445v1.666H8.765V15H4.166v-4.95z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M10 5.833v0A4.167 4.167 0 005.833 10v.834"
            opacity="0.3"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
      <span
        class="bp3-button-text"
      >
        notifications.name
      </span>
    </button>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main iconButton flowControlIcon StyledProps--font StyledProps--font-weight-semi-bold StyledProps--font-size-xsmall Button--withLeftIcon Button--variation Button--variation-tertiary"
      type="button"
    >
      <span
        class="bp3-icon bp3-icon-settings StyledProps--main StyledProps--padding-right-xsmall"
        icon="settings"
      >
        <svg
          data-icon="settings"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <desc>
            settings
          </desc>
          <path
            d="M4 1c0-.55-.45-1-1-1S2 .45 2 1v5h2V1zM2 19c0 .55.45 1 1 1s1-.45 1-1v-6H2v6zm9-18c0-.55-.45-1-1-1S9 .45 9 1v8h2V1zm7 0c0-.55-.45-1-1-1s-1 .45-1 1v3h2V1zM9 19c0 .55.45 1 1 1s1-.45 1-1v-3H9v3zm9-14h-2c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm-2 14c0 .55.45 1 1 1s1-.45 1-1v-8h-2v8zM4 7H2c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V8c0-.55-.45-1-1-1zm7 3H9c-.55 0-1 .45-1 1v3c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-3c0-.55-.45-1-1-1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="bp3-button-text"
      >
        pipeline.barriers.flowControl
      </span>
    </button>
    <div />
    <div />
  </div>
</div>
`;
